src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js

```diff
@@ -76,12 +80,25 @@ class BasePaymentsPage extends React.Component {
             this.setMenuPosition();
         }
 
+        // If the user was previously offline, skip debouncing showing the loader
+        if (prevProps.network.isOffline && !this.props.network.isOffline) {
+            this.setShouldShowLoadingSpinner();
+        } else {
+            this.debounceSetShouldShowLoadingSpinner();
+        }
+
+        // previously online OR currently offline, skip fetch
         if (!prevProps.network.isOffline || this.props.network.isOffline) {
             return;
         }
+
         this.fetchData();
     }
 
+    setShouldShowLoadingSpinner() {
```

Comment:
This function is being called in a loop. Every time we fetchData we update the component, then we call this function, then we fetch data again.